### PR TITLE
IOP DMAC cleanup

### PR DESCRIPTION
**Pull request by kernle32dll**
Corrects the IOP DMAC addresses for SIO2 in/out, and moves the channel IDs for SPU2 and DEV9.

Should be pretty understandable if viewed per-commit.

### CODE REVIEW

File: Source/iop/Iop_Dmac.cpp [hunk 1]
```diff
@@ -19,11 +19,11 @@ using namespace Iop::Dmac;
 CDmac::CDmac(uint8* ram, CIntc& intc)
     : m_ram(ram)
     , m_intc(intc)
-    , m_channelSpu0(CH4_BASE, CHANNEL_SPU0, *this)
-    , m_channelSpu1(CH8_BASE, CHANNEL_SPU1, *this)
-    , m_channelDev9(CH9_BASE, CHANNEL_DEV9, *this)
-    , m_channelSio2In(CH11_BASE, CHANNEL_SIO2in, *this)
-    , m_channelSio2Out(CH12_BASE, CHANNEL_SIO2out, *this)
+    , m_channelSpu0(CH4_BASE, CHANNEL_SPU0, Iop::CIntc::LINE_DMA4, *this)
+    , m_channelSpu1(CH7_BASE, CHANNEL_SPU1, Iop::CIntc::LINE_DMA8, *this)
```
Comment (jpd002) [inline Source/iop/Iop_Dmac.cpp:23]: While we're at it, I think we should rename `LINE_DMA4` and `LINE_DMA8` to `LINE_DMA_SPU0` and `LINE_DMA_SPU1`

File: Source/iop/Iop_Intc.h [hunk 2]
```diff
@@ -36,6 +36,8 @@ namespace Iop
 			LINE_DMA_DEV9 = 0x29,
 			LINE_DMA9 = 0x2A,  //sif0
 			LINE_DMA10 = 0x2B, //sif1
+			LINE_DMA11 = 0x2C, // SIO2i
```
Comment (jpd002) [inline Source/iop/Iop_Intc.h:39]: Let's name these `LINE_DMA_SIO2in` and `LINE_DMA_SIO2out`.

File: Source/iop/Iop_Dmac.h [hunk 3]
```diff
@@ -31,10 +31,10 @@ namespace Iop
 			CH4_BASE = 0x1F8010C0,
 			CH5_BASE = 0x1F8010D0,
 			CH6_BASE = 0x1F8010E0,
-			CH8_BASE = 0x1F801500,
-			CH9_BASE = 0x1F801510,
-			CH11_BASE = 0x1F801530, //Unsure about that
-			CH12_BASE = 0x1F801540, //Unsure about that
+			CH7_BASE = 0x1F801500,
+			CH8_BASE = 0x1F801510,
+			CH11_BASE = 0x1F801540, //Unsure about that
```
Comment (jpd002) [inline Source/iop/Iop_Dmac.h:36]: We should get rid of the comments that say we're unsure :)

Comment (kernle32dll) [reply Source/iop/Iop_Dmac.h]: Done :D

File: Source/iop/Iop_Intc.h (hunk 2 above)
Comment (kernle32dll) [reply Source/iop/Iop_Intc.h]: Done

File: Source/iop/Iop_Dmac.cpp (hunk 1 above)
Comment (kernle32dll) [reply Source/iop/Iop_Dmac.cpp]: Done

**Comment by jpd002** [comment]
Awesome, thanks for cleaning these things up!

**Pull request merged**